platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/problems/ConfigurationCacheProblems.kt

```diff
@@ -164,9 +175,32 @@ class ConfigurationCacheProblems(
         }
     }
 
+    override fun forTask(task: Task): ProblemsListener {
+        val taskDegradationReasons = degradationController.degradationReasons
+            .filterIsInstance<DegradationReason.Task>()
```

Comment:
🚀 If similar logic remains in the final implementation it should be O(1)

Reply:
Addressed in the new PR
https://github.com/gradle/gradle/pull/33740/files#diff-dd265cff99d54678fb48066f9ad6acf43867ee9e4355909f517983b58b4541eaR188